gtm/README.md

```diff
@@ -82,6 +82,42 @@ instantsearch.widgets.hits({
 > - Vue InstantSearch 2.1.0
 > - Angular InstantSearch 3.0.0
 
+Once these values are available in the hits template, you can retrieve them in GTM.
+
+1. Create a Custom JavaScript variable to find the element and name it `Find closest`
+
+```js
+function() {
+  return function(target, selector) {
+    while (!target.matches(selector) && !target.matches('body')) {
+      target = target.parentElement;
+    }
+
+    return target.matches(selector) ? target : undefined;
+  }
+}
+```
+
+This creates a variable called `{{Find closest}}` that contains the nearest element matching the selector.
+
+2. Create another Custom JavaScript variable and name it `ObjectIDClicked`
+
+```js
+function() {
+  var element = {{Find closest}}({{Click Element}}, '[data-insights-object-id]');
+
+  return typeof element !== 'undefined'
+    ? element.getAttribute('data-insights-object-id')
+    : undefined;
+}
+```
+
+This creates a variable called `{{ObjectIDClicked}}` that gives you the object ID of the hit clicked to reference it in the Insights method.
```

Comment:
Thanks for this update françois. 
Shoudn't there be a variable of `queryID`, `eventName`, `position`, `index`?
A analytics click required all the following 
```js
{
  eventType: "click",
  eventName: "Add to cart",
  objectIDs: ["xxx"],
  positions: [2],
  queryID: "xxx",
  index: "xxx",
  userToken: "xxx"
}


```
Is it implied that a variable should be created for each of thses ? (expect maybe for the user token )

Reply:
It's implied but you're right, I'm gonna mention that.

Note that in my GTM example, I created variables for `ObjectIDClicked`, `ObjectIDPosition` and `QueryID`.

`index` and `eventName` were provided in the GTM interface. It would make sense to create a variable for `index` actually, but not sure about `eventName`.

`userToken` is its own "Data Layer Variable".